### DISCUSSION TITLE
Model Selection

### DISCUSSION THREAD

**Discussion by mosquitobird11** [hlky/stable-diffusion-webui#672] [Feature suggestions]
On September 7 (in two days from now) [Haru](https://github.com/harubaru/waifu-diffusion/) is going to release a finetuned SD1.4 model for anime-like generations, trained on a set of 56,000 high aesthetic CLIP-filtered Danbooru images. 

This, along with the suspected 1.5 SD coming in the near future, would be more accessible if we had a way to easily switch between models in the webUI, even though this would require releasing and reloading the new model into memory.

**Comment by Muskelmagier** [comment]
Yeah, with another model i follow  being released in the near future, this becomes even higher priority

**Comment by lendrick** [comment]
The waifu diffusion model can be downloaded here if anyone is working on this and needs another model to test with:

https://storage.googleapis.com/ws-store2/wd-v1-2-full-ema.ckpt

**Comment by mosquitobird11** [comment]
@Muskelmagier @lendrick and anyone else who wants a quick fix for this, this is what I did for now:

in webui.py, at the beginning of `load_model_from_config` I added the lines:
```    
    all_models = {
        "1": "models/ldm/stable-diffusion-v1/model.ckpt",
        "2": "models/ldm/stable-diffusion-v1/model_waifu1.2.ckpt",
    }

    while 1:
        modelnum = input('What model number?\nEnter 1 for: Stable Diffusion 1.4\nEnter 2 for: Waifu Diffusion 1.2\n')
        if modelnum in all_models:
            ckpt = all_models[modelnum]
            break
 ```
 
Obviously you have to have models in that location named like that for it to make sense, but this is pretty extensible and easy for me to choose a model on boot without renaming models etc.

**Reply by Batman313v** [reply]
So if this is the before:
![image](https://user-images.githubusercontent.com/25057949/189471795-12e21ebd-e03e-4e1f-b3d5-3ff2e49108b5.png)

You pasted it here?
![image](https://user-images.githubusercontent.com/25057949/189471815-57c321bb-6b08-4f01-868d-abc6be346eb5.png)

**Reply by mosquitobird11** [reply]
Yeah that's correct! The code is super hacky / simplistic but it should be easy to follow :)
